network/network.go

```diff
@@ -71,11 +71,11 @@ func loadOrCreateKey(path string) (lp2pcrypto.PrivKey, error) {
 	return key, nil
 }
 
-func NewNetwork(conf *Config) (Network, error) {
-	return newNetwork(conf, []lp2p.Option{})
+func NewNetwork(ctx context.Context, cancel func(), conf *Config) (Network, error) {
```

Comment:
You should not pass cancel to the Network module.

Reply:
but, the network module needs the cancel!
should I remove it from the Network module completely?

Reply:
Why network module needs cancel?
This context is global and should be created when the node starts and cancelled when the node finishes.

Reply:
so, that's means I should remove the network cancel and add it to the node struct and call it when the node stopped.
right?